### DISCUSSION TITLE
linkerd-crds & linkerd-control-plane version relationship?

### DISCUSSION THREAD

**Discussion by bjthomas1** [linkerd/linkerd2#11159] [Q&A]
Is there a table in any Linkerd documentation anywhere that describes which version of Helm Chart `linkerd-crds` should be used with `linkerd-control-plane`?

Currently my cluster's setup looks something like:-

```
+-----------------------+-----------------+-----------------------+-------------+
| Linkerd Component     | Current Version | Helm Chart            | Helm Chart  |
| Name                  |                 | Name                  | Version     |
+-----------------------+-----------------+-----------------------+-------------+
| linkerd-destination   |                 |                       |             |
| linkerd-identity      | stable-2.13.5   | linkerd-control-plane | 1.12.5      |
| linkerd-proxy-injector|                 |                       |             |
+-----------------------+-----------------+-----------------------+-------------+
| various linkerd CRDs  |        ?        | linkerd-crds          | 1.6.1       |
+-----------------------+-----------------+-----------------------+-------------+
```

I deploy `linkerd-crds` & `linkerd-control-plane` as two separate Helm charts in their own deployment pipelines (because CRDs must be installed before the rest of Linkerd). I would like to automatically deploy the latest minor release (e.g. stable-2.13.*) which I'd do with my own "Parent" Helm chart, e.g

Chart.yaml:-
```
dependencies:
  - name: linkerd-control-plane
    version: ~1.12
    repository: https://helm.linkerd.io/stable
    condition: linkerd-control-plane.enabled
    alias: linkerdControlPlane
```

where version: ~1.12 should fetch the latest Helm Chart 1.12.5, which contains the appVersion stable-2.13.5. The recent `linkerd-control-plane` Helm Chart versions appear to marry-up with Linkerd app versions like:-

```
+------------+--------+
|Helm Chart  | App    |
+------------+--------+
| 1.12.*     | 2.13.* |
| 1.9.*      | 2.12.* |
+------------+--------+
```

but I'm not sure about the `linkerd-crds` Chart versions, and how they relate to the main application Chart. For example, does:-

```
+--------------+--------+
|linkerd-crds  | linkerd-control-plane |
+--------------+--------+
| 1.6.1        | 1.12.5 |
| 1.6.0        | 1.12.4 |
|              | 1.12.3 |
|              | 1.12.2 |
|              | 1.12.1 |
+--------------+--------+
| 1.4.0        | 1.9.7  |
+--------------+--------|
```

relate to one-another?

**tl;dr** - how do I know/handle the major version changes in linkerd-crds such that I don't break them with a wrong linkerd-control-plane ?

**Comment by doslindos** [comment]
Here's a table mapping the relations on latest linkerd versions https://linkerd.io/releases/#helm-chart-version-matrix

**Reply by florian-besser** [reply]
This link no longer seems to contain such a matrix.

I believe this has been moved to https://linkerd.io/2.15/reference/helm-chart-version-matrix/
